### PR TITLE
feat: Chat History and cosmosdb implementation

### PR DESCRIPTION
**Pull request by Fr4nc3**
## Purpose

This pull request introduces the addition of chat history to the Chat With Your Data (CWYD) website, leveraging Cosmos DB for persistent storage of user conversations. This allows users to access past interactions, making the platform more user-friendly and improving the continuity of conversations. It also helps to refine the AI responses by referencing previous queries.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## How to Test
*  Get the code
- Download this PR and install using command line azd up
- go to the CWYD new frontend enviroment 
- Check the new Chat history features
-
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
- Chat history is stored correctly in Cosmos DB.
- Users can retrieve and view past chats.
- Performance remains consistent after chat history is enabled.

## Other Information

This update integrates smoothly with existing infrastructure without altering the core chat functionalities, allowing for a seamless experience for both users and administrators.

### CODE REVIEW

File: .vscode/launch.json
Comment (Fr4nc3) [inline .vscode/launch.json:1]: this need to be rollback before final PR

File: Pipfile
Comment (Fr4nc3) [inline Pipfile:1]: This need to be git ignored before final PR

File: Pipfile.lock
Comment (Fr4nc3) [inline Pipfile.lock:1]: This need to be gitignore before the final PR

File: code/backend/batch/function_app.py [hunk 1]
```diff
@@ -14,9 +15,10 @@
     configure_azure_monitor()
 
 app = func.FunctionApp(
-    http_auth_level=func.AuthLevel.FUNCTION
+    http_auth_level=func.AuthLevel.ANONYMOUS,
```
Comment (Fr4nc3) [inline code/backend/batch/function_app.py:18]: this need to be rolled back before final PR

File: code/backend/batch/utilities/helpers/config/config_helper.py [hunk 2]
```diff
@@ -176,21 +176,7 @@ def _set_new_config_properties(config: dict, default_config: dict):
     @staticmethod
     @functools.cache
     def get_active_config_or_default():
-        env_helper = EnvHelper()
```
Comment (Fr4nc3) [inline code/backend/batch/utilities/helpers/config/config_helper.py:179]: this needs to be rolled back before final PR

File: code/backend/batch/utilities/helpers/env_helper.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def __new__(cls):
 
     def __load_config(self, **kwargs) -> None:
         load_dotenv()
-
+        print(os.environ)
```
Comment (Fr4nc3) [inline code/backend/batch/utilities/helpers/env_helper.py:26]: this needs to be rolled back before final PR

File: code/frontend/vite.config.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export default defineConfig({
         host: true,
         proxy: {
             "/api": {
-                target: "http://127.0.0.1:5050",
```
Comment (Fr4nc3) [inline code/frontend/vite.config.ts:16]: this need to be rolled back before final PR

**Review by marktayl1 — COMMENTED** [review]
Just found the pip file thing.

**Review by marktayl1 — APPROVED** [review]
Test approval per Francia's request.  Do not merge.

**Review by marktayl1 — CHANGES_REQUESTED** [review]
test

**Review by marktayl1 — APPROVED** [review]

**Review by Vinay-Microsoft — APPROVED** [review]

**Review by Vinay-Microsoft — APPROVED** [review]

**Comment by ross-p-smith** [comment]
@Fr4nc3, @Vinay-Microsoft - this PR seems like it has commented out tests and linting - do you have a plan to put these back?

**Comment by ross-p-smith** [comment]
Closes #85

**Pull request merged**